src/platforms/esp32/components/avm_sys/platform_nifs.c

```diff
@@ -267,13 +285,77 @@ static term nif_esp_sleep_get_wakeup_cause(Context *ctx, int argc, term argv[])
     switch (cause) {
         case ESP_SLEEP_WAKEUP_UNDEFINED:
             return UNDEFINED_ATOM;
+        case ESP_SLEEP_WAKEUP_EXT0:
+            return context_make_atom(ctx, sleep_wakeup_ext0_atom);
+        case ESP_SLEEP_WAKEUP_EXT1:
+            return context_make_atom(ctx, sleep_wakeup_ext1_atom);
         case ESP_SLEEP_WAKEUP_TIMER:
             return context_make_atom(ctx, sleep_wakeup_timer_atom);
+        case ESP_SLEEP_WAKEUP_TOUCHPAD:
+            return context_make_atom(ctx, sleep_wakeup_touchpad_atom);
+        case ESP_SLEEP_WAKEUP_ULP:
+            return context_make_atom(ctx, sleep_wakeup_ulp_atom);
+        case ESP_SLEEP_WAKEUP_GPIO:
+            return context_make_atom(ctx, sleep_wakeup_gpio_atom);
+        case ESP_SLEEP_WAKEUP_UART:
+            return context_make_atom(ctx, sleep_wakeup_uart_atom);
+#ifdef ESP_SLEEP_WAKEUP_WIFI
+        case ESP_SLEEP_WAKEUP_WIFI:
+            return context_make_atom(ctx, sleep_wakeup_wifi_atom);
+#endif
+#ifdef ESP_SLEEP_WAKEUP_COCPU
+        case ESP_SLEEP_WAKEUP_COCPU:
+            return context_make_atom(ctx, sleep_wakeup_cocpu_atom);
+#endif
+#ifdef ESP_SLEEP_WAKEUP_COCPU_TRAP_TRIG
+        case ESP_SLEEP_WAKEUP_COCPU_TRAP_TRIG:
+            return context_make_atom(ctx, sleep_wakeup_cocpu_trap_trig_atom);
+#endif
+#ifdef ESP_SLEEP_WAKEUP_BT
+        case ESP_SLEEP_WAKEUP_BT:
+            return context_make_atom(ctx, sleep_wakeup_bt_atom);
+#endif
         default:
             return ERROR_ATOM;
     }
 }
 
+static term nif_esp_sleep_enable_ext0_wakeup(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+
+    VALIDATE_VALUE(argv[0], term_is_integer);
+    VALIDATE_VALUE(argv[1], term_is_integer);
+    gpio_num_t pin = term_to_int(argv[0]);
+    int level = term_to_int(argv[1]);
+    esp_err_t err = esp_sleep_enable_ext0_wakeup(pin, level);
```

Comment:
I think this is fine, because `esp_sleep_enable_ext*_wakeup` will catch the case where level is outside of the range of `0..1`, at least according to the docs.